src/external/electrs.ts

```diff
@@ -224,11 +237,33 @@ export class DefaultElectrsAPI implements ElectrsAPI {
         return amount;
     }
 
+    async getEarliestPaymentToRecipientAddressTxId(recipientAddress: string, amount?: BitcoinAmount): Promise<string> {
+        try {
+            const txes = await this.getData(this.addressApi.getAddressTxHistory(recipientAddress));
+            if (txes.length >= 25) {
+                throw new Error("Over 25 transactions returned; this is either a highly non-standard vault, or not a vault address");
+            }
+
+            const oldestTx = txes.pop();
+            if (!oldestTx || !(oldestTx.vout)) {
+                throw new Error("No transaction found for recipient and amount");
+            }
+            for (const txo of oldestTx.vout) {
+                if (txo.scriptpubkey_address === recipientAddress && this.txoHasAtLeastAmount(txo, amount)) {
+                    return oldestTx.txid;
+                }
+            }
+        } catch (e) {
+            return Promise.reject(new Error(`Error during tx lookup by address: ${e}`));
+        }
+        return Promise.reject(new Error("No transaction found for recipient and amount"));
+    }
+
     async getUtxoTxIdByRecipientAddress(recipientAddress: string, amount?: BitcoinAmount): Promise<string> {
```

Comment:
Is this still used? I can't foresee where we would still need to fetch the UTXOs.